plugin/evm/warp_client.go

```diff
@@ -0,0 +1,57 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/subnet-evm/rpc"
+
+	"github.com/ava-labs/avalanchego/utils/cb58"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// Interface compliance
+var _ WarpClient = (*warpClient)(nil)
+
+type WarpClient interface {
+	GetSignature(ctx context.Context, signatureRequest message.SignatureRequest) (*[bls.SignatureLen]byte, error)
+}
+
+// warpClient implementation for interacting with EVM [chain]
+type warpClient struct {
+	client *rpc.Client
+}
+
+// NewWarpClient returns a WarpClient for interacting with EVM [chain]
+func NewWarpClient(uri, chain string) (WarpClient, error) {
+	client, err := rpc.Dial(fmt.Sprintf("%s/ext/bc/%s/rpc", uri, chain))
+	if err != nil {
+		log.Error("failed to dial client")
```

Comment:
Can we remove all of the logs from this file in favor of returning wrapped errors with the necessary details to the caller?